client/my-sites/site-settings/form-general.jsx

```diff
@@ -113,57 +119,136 @@ const FormGeneral = React.createClass( {
 			amp_is_supported: false,
 			amp_is_enabled: false,
 		} );
-	},
+		this.props.clearDirtyFields();
+		this.props.updateFields( this.getFormSettings( this.props.settings ) );
+	}
 
-	onTimezoneSelect( timezone ) {
-		this.setState( { timezone_string: timezone } );
-	},
+	componentWillReceiveProps( nextProps ) {
+		this._showWarning( nextProps.site );
+
+		if ( nextProps.settings !== this.props.settings ) {
+			let newState = this.getFormSettings( nextProps.settings );
+			//If we have any fields that the user has updated,
+			//do not wipe out those fields from the poll update.
+			newState = omit( newState, nextProps.dirtyFields );
+			nextProps.updateFields( newState );
+		}
+
+		if (
+			this.props.isSavingSettings &&
+			! nextProps.isSavingSettings &&
+			nextProps.isSaveRequestSuccessful
+		) {
+			nextProps.clearDirtyFields();
+			nextProps.markSaved();
+		}
+	}
 
 	onRecordEvent( eventAction ) {
 		return this.recordEvent.bind( this, eventAction );
-	},
+	}
 
 	onRecordEventOnce( key, eventAction ) {
 		return this.recordEventOnce.bind( this, key, eventAction );
-	},
+	}
+
+	recordEvent( eventAction ) {
+		debug( 'record event: %o', eventAction );
+		analytics.ga.recordEvent( 'Site Settings', eventAction );
+	}
+
+	/**
+	 * Record an analytics event only once per mounted component instance
+	 * @param  {string} key         - unique key to namespace the event
+	 * @param  {string} eventAction - the description of the action to appear in analytics
+	 */
+	recordEventOnce( key, eventAction ) {
+		debug( 'record event once: %o - %o', key, eventAction );
+		if ( this.state[ 'recordEventOnce-' + key ] ) {
+			return;
+		}
+		this.recordEvent( eventAction );
+		this.setState( { [ 'recordEventOnce-' + key ]: true } );
+	}
+
+	handleRadio = event => {
+		const currentTargetName = event.currentTarget.name,
+			currentTargetValue = event.currentTarget.value;
+
+		this.props.updateFields( { [ currentTargetName ]: currentTargetValue } );
+	};
+
+	handleSubmitForm = event => {
+		if ( ! event.isDefaultPrevented() && event.nativeEvent ) {
+			event.preventDefault();
+		}
+
+		this.submitForm();
+		this.recordEvent( 'Clicked Save Settings Button' );
+	};
+
+	submitForm() {
```

Comment:
I fixed some remaining issues on ff97425 and I tried to reduxify another settings form on https://github.com/Automattic/wp-calypso/pull/9899/commits/1df5abe7e0d683448792f1749a111620c54a6444 and see what can be factorized. 

So after this my feelings are:
 - We should duplicate the `handleRadio`, `handleCheckbox` functions because they are UI related and they can be written in different ways
 - We may want to group `connectComponent`, `componentWillReceiveProps`, `componentWillMount`, `submitForm` in a `siteSettingsForm` HoC or base class (Yes, I changed my mind :) )
 - wait for #9867 and use the redux middleware for recording events

I also think we should merge this as is, and factorize, on the next PR #9899 (reduxifying a second form).

Thoughts ?